File: seed/php-sdk/simple-fhir/src/Types/Account.php
<?php

namespace Seed\Types;

use Seed\Core\SerializableType;
use Seed\Core\JsonProperty;

class Account extends SerializableType
{
    /**
     * @var string $resourceType
     */
    #[JsonProperty("resource_type")]
    public string $resourceType;

    /**
     * @var string $name
     */
    #[JsonProperty("name")]
    public string $name;

    /**
     * @var ?Patient $patient
     */
    #[JsonProperty("patient")]
    public ?Patient $patient;

    /**
     * @var ?Practitioner $practitioner
     */
    #[JsonProperty("practitioner")]
    public ?Practitioner $practitioner;

    /**
     * @param string $resourceType
     * @param string $name
     * @param ?Patient $patient
     * @param ?Practitioner $practitioner
     */
    public function __construct(
        string $resourceType,
        string $name,
        ?Patient $patient = null,
        ?Practitioner $practitioner = null,
    ) {
        $this->resourceType = $resourceType;
        $this->name = $name;
        $this->patient = $patient;
        $this->practitioner = $practitioner;
    }
}

File: seed/ruby-sdk/simple-fhir/lib/seed/types/account.rb
# frozen_string_literal: true

module Seed
  module Types
    class Account < Internal::Types::Model
      field :resource_type, -> { String }, optional: false, nullable: false
      field :name, -> { String }, optional: false, nullable: false
      field :patient, -> { Seed::Types::Patient }, optional: true, nullable: false
      field :practitioner, -> { Seed::Types::Practitioner }, optional: true, nullable: false
    end
  end
end
